maas/data_source_maas_boot_resource_test.go

```diff
@@ -0,0 +1,144 @@
+			{
+				Config: testAccDataSourceMaasBootResources(os, release, arches, subarches, labels),
+				Check:  resource.ComposeTestCheckFunc(checks...),
+			},
+		},
+	})
+}
+
+func testAccDataSourceMaasBootResources(os string, release string, arches []string, subarches []string, labels []string) string {
+	return fmt.Sprintf(`
+%s
+
+data "maas_boot_resources" "test" {
+	os      = maas_boot_source_selection.test.os
+	release = maas_boot_source_selection.test.release
+}
+`, testAccMaasBootSourceSelection(os, release, arches, subarches, labels))
+}
+
+func testAccCheckDataSourceMaasBootResourcesDestroy(s *terraform.State) error {
+	// retrieve the connection established in Provider configuration
+	conn := testutils.TestAccProvider.Meta().(*maas.ClientConfig).Client
+
+	// loop through the resources in state
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "maas_boot_resources" {
+			continue
+		}
+
+		// fetch the boot source
+		bootsource, err := conn.BootSources.Get()
+		if err != nil {
+			return fmt.Errorf("error fetching boot sources: %v", err)
+		}
+		boot_source_id := bootsource[0].ID
+
+		// fetch all the synced resources
+		response, err := conn.BootResources.Get(&entity.BootResourcesReadParams{Type: "synced"})
+		if err != nil {
+			return fmt.Errorf("error getting synced boot resource: %s", err)
+		}
+		resourceMap := make(map[string]struct{})
+		for _, res := range response {
+			resourceMap[res.Name] = struct{}{}
+		}
+
+		// we need to read each resource seperately
+		count := rs.Primary.Attributes["boot_resources.#"]
+		selectionCount, err := strconv.Atoi(count)
+		if err != nil {
+			return fmt.Errorf("Could not convert %v to integer: %v", count, err)
+		}
+		if selectionCount < 1 {
+			return fmt.Errorf("Boot Resource does not contain any selections!")
+		}
+
+		// check each resource has been deleted
+		for i := 0; i < selectionCount; i++ {
+			this_name := rs.Primary.Attributes[fmt.Sprintf("boot_resources.%d.name", i)]
+			if _, exists := resourceMap[this_name]; exists {
+				return fmt.Errorf("Boot Resource still exists for %s", this_name)
+			}
+
+			parts := strings.SplitN(this_name, "/", 2)
+			if len(parts) < 2 {
+				return fmt.Errorf("Invalid resource name: %s", this_name)
+			}
+			os, release := parts[0], parts[1]
+
+			if bootsourceselection, err := findBootSourceSelection(conn, boot_source_id, os, release); err != nil {
+				// 404 means the resource was deleted already
+				if !strings.Contains(err.Error(), "404 Not Found") {
+					continue
+				}
+				// anything else is an error
+				return fmt.Errorf("error finding selection '%v': %v", this_name, err)
+			} else if bootsourceselection != nil {
+				return fmt.Errorf("boot source selection (%s) was unexpectedly found on deleted resource", this_name)
+			}
+		}
+		return nil
+	}
+
+	return nil
+}
+
+func findBootSourceSelection(client *client.Client, boot_source int, os string, release string) (*entity.BootSourceSelection, error) {
```

Comment:
what about moving this to `resource_maas_boot_resource.go` like we do with others?